src/models/schema.ts

```diff
@@ -96,6 +111,54 @@ export class Schema extends Data implements IResourceBase {
   }
 }
 
+/**
+ * Thin ISearchable wrapper around a schema subject string. Needs to carry the metadata
+ * from its source schema registry with it for later API calls.
+ */
+export class Subject implements IResourceBase, ISearchable {
+  connectionId!: ConnectionId;
+  environmentId!: EnvironmentId;
+  schemaRegistryId!: string;
+
+  name!: string;
+  id!: string;
```

Comment:
Is there any reason we can't use `schemas: Schema[] = []` and an optional `type` for when we handle the preloading so we don't have to use `ContainerTreeItem<Schema>` in the Topics view? I feel like the `Subjectish` split between `Subject` and `ContainerTreeItem<Schema>` is making this more complicated since we use them in the same way in each view (but the Topics view has the `KafkaTopic` parent items). 🤔

Reply:
Yes, will go down this route and see what it looks like. I agree.